From 1f6bb0d3f6e44bece655620cdc55083d9645bb2b Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Thu, 27 Jul 2017 22:24:13 +0800
Subject: [PATCH 2302/5242] MLK-16013-23 staging: typec: tcpm: Do soft reset
 for unexpected get sink cap

commit  8e0d4bef56919fdd834b1d9047fca355da9816e9 from
https://source.codeaurora.org/external/imx/linux-imx.git

A source or sink should receive the get_sink_cap message when in SRC_RDY or
SNK_RDY, if the port receives a get sink capability request before ready,
which means there is message sync error, we should do soft reset to recover
from it, instead of sending a reject message.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/typec/tcpm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/typec/tcpm.c b/drivers/usb/typec/tcpm.c
index b55ebb7..627f308 100644
--- a/drivers/usb/typec/tcpm.c
+++ b/drivers/usb/typec/tcpm.c
@@ -1595,7 +1595,7 @@ static void tcpm_pd_ctrl_request(struct tcpm_port *port,
 			tcpm_queue_message(port, PD_MSG_DATA_SINK_CAP);
 			break;
 		default:
-			tcpm_queue_message(port, PD_MSG_CTRL_REJECT);
+			tcpm_set_state(port, SOFT_RESET_SEND, 0);
 			break;
 		}
 		break;
-- 
1.7.9.5

